### DISCUSSION TITLE
QCODES FPGA

### DISCUSSION THREAD

**Discussion by fll409** [microsoft/Qcodes#5457] [Q&A] [answered]
Hello QCoDes Teams:

    I am newer to QCODES project, I already read the https://qcodes.github.io/Qcodes/community/index.html carefully, 
    and know how qcodes to work, there is a requirement about use qcodes to work with FPGA, and use it, 
    but I don't find the driver about FPGA in qcodes_contrib_drivers, 
    does it means I need write the driver about FPGA by myself ? 
    And below are my questions:
    1. Do I need write a driver about FPGA like driver in https://github.com/QCoDeS/Qcodes_contrib_drivers ? 
       If yes, so I have to get driver program from FGPA vendor, right?
    2. And when I write FPGA driver, I need Follow the requirements of below document 
       https://qcodes.github.io/Qcodes/examples/index.html#writing-drivers, right?
    3. Or are there any suggestions about how qcodes work with FPGA ? Any suggestions will be appropriate.
    
Thanks in advances
Fred

**Comment by fll409** [comment]
Thanks for any feedback

**Reply by jenshnielsen** [reply, marked as answer]
Hi Fred, 

There are no FPGA specific drivers in QCoDeS. In general, I don't really think that python is a suitable environment for programming FPGAs. Typically, FPGAs are programmed using HDL https://en.wikipedia.org/wiki/Field-programmable_gate_array#Programming or something similar. That being said if you want to interface with a specific FPGA that is already programmed that may be possible but how to do it would depend entirely on which device you are programming.
